ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py

```diff
@@ -0,0 +1,970 @@
+                                       self.module, 1, 'checkQueuesButRabbitIsWorking', False)
+        else:
+            self.LOGGER.logger("OVS workers are UP! Maybe it is stuck? Start checking the queues...", self.module, 0,
+                               'checkQueuesButOvsWorkersAreUp', False)
+
+        # fetch remaining tasks on node
+        self.LOGGER.logger("Starting deep check for RabbitMQ!", self.module, 3, 'DeepCheckRabbit', False)
+
+        #
+        # Rabbitmq check: queue verification
+        #
+        rcommand = "rabbitmqctl list_queues | grep ovs_ | sed -e 's/[[:space:]]\+/ /g' | cut -d ' ' -f 2"
+        output_01 = self.utility.execute_bash_command(rcommand)
+        time.sleep(15)
+        output_02 = self.utility.execute_bash_command(rcommand)
+
+        # check diff/results and continue
+        lost_queues = []
+        if set(output_01) - set(output_02):
+            # found some changed queue's!
+            for i, j in zip(output_01, output_02):
+                if int(i) < int(j):
+                    # queue is building up
+                    self.LOGGER.logger(
+                        "Seems like queue '{0}' is building up! Please verify this with 'rabbitmqctl list_queues "
+                        "| grep ovs_'".format(output_02.index(j)), self.module, 2, 'process_celery_queue_{0}'
+                        .format(output_02.index(j)), False)
+                    lost_queues.append(output_02.index(j))
+                elif int(i) > int(j):
+                    # queue is decreasing
+                    self.LOGGER.logger("Seems like queue '{0}' is working fine!".format(output_02.index(j)),
+                                       self.module, 1, 'process_celery_queue_{0}'.format(output_02.index(j)), False)
+            # post-check
+            if len(lost_queues) > 0:
+                return False
+            else:
+                return True
+        else:
+            self.LOGGER.logger(
+                "Seems like all Celery queues are stuck, you should check: 'rabbitmqctl list_queues' and 'ovs-workers'",
+                self.module, 0, 'process_celery_all_queues', False)
+            return False
+
+    def check_ovs_workers(self):
+        """
+        Extended check of the Open vStorage workers; When the simple check fails, it will execute a full/deep check.
+        """
+
+        self.LOGGER.logger("Checking if OVS-WORKERS are running smoothly: ", self.module, 3, 'check_ovs_workers', False)
+
+        # init timout for x amount of sec for celery
+        signal.signal(signal.SIGALRM, self._method_handler)
+        signal.alarm(7)
+
+        # checking celery
+        try:
+            # basic celery check
+            self._check_celery()
+            self.LOGGER.logger("The OVS-WORKERS are working smoothly!", self.module, 1, 'process_celery')
+        except Exception, ex:
+            # kill alarm
+            signal.alarm(0)
```

Comment:
Right, I now see that the alarm is reset in the ```self._check_celery()``` method. This is quite cumbersome, since it could introduce bugs when people are changing code. You should reset the alarm where you armed it. Besides, does it makes sense to have a separate method for checking celery? Why not incorporate it in this method?